src/libstd/fs.rs

```diff
@@ -0,0 +1,1474 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Filesystem manipulation operations
+//!
+//! This module contains basic methods to manipulate the contents of the local
+//! filesystem. All methods in this module operate reasonable across all
+//! supported platforms, and extra functionality for platform-specific features
+//! can be found in the extension traits of `std::os::$platform`.
+
+#![unstable(feature = "fs")]
+
+use core::prelude::*;
+
+use io::{self, Error, ErrorKind, SeekFrom, Seek, Read, Write};
+use path::{AsPath, Path, PathBuf};
+use sys::fs2 as fs_imp;
+use sys_common::{AsInnerMut, FromInner};
+use vec::Vec;
+
+/// A reference to an open file on the filesystem.
+///
+/// An instance of a `File` can be read and/or written depending on what options
+/// it was opened with. Files also implement `Seek` to alter the logical cursor
+/// that the file contains internally.
+///
+/// # Example
+///
+/// ```no_run
+/// use std::io::prelude::*;
+/// use std::fs::File;
+///
+/// let mut f = File::create("foo.txt").unwrap();
+/// f.write_all(b"Hello, world!").unwrap();
+///
+/// let mut f = File::open("foo.txt").unwrap();
+/// let mut s = String::new();
+/// f.read_to_string(&mut s).unwrap();
+/// assert_eq!(s, "Hello, world!");
+/// ```
+pub struct File {
+    inner: fs_imp::File,
+    path: PathBuf,
+}
+
+/// Metadata information about a file.
+///
+/// This structure is returned from the `file_attr` function or method and
+/// represents known metadata about a file such as its permissions, size,
+/// modification times, etc.
+pub struct FileAttr(fs_imp::FileAttr);
```

Comment:
I still prefer `Meta` or `Metadata` here (and seems like the RFC thread is ok with that?)